### ISSUE TITLE
organizing executors by system

### ISSUE THREAD

**Issue by shahzebsiddiqui** [buildtesters/buildtest#656]
Currently there is no grouping of executors by system name. Current implementation organizes executors by their type but it doesn't tell where executors should be run.

Currently executors are defined like

```
executors:
  local:
    bash:
      description: submit jobs on local machine using bash shell
      shell: bash
    csh:
      description: submit jobs on local machine using csh shell
      shell: csh
    python:
      description: submit jobs on local machine using python shell
      shell: python
    sh:
      description: submit jobs on local machine using sh shell
      shell: sh
    zsh:
      description: submit jobs on local machine using zsh shell
      shell: zsh
```

One idea is to group them with one level hierarchy organized by systems.

```
executors:
  system1:
    host: [ login1, login2, login3 ]
    local:
      bash:
        description: submit jobs on local machine using bash shell
        shell: bash
      csh:
        description: submit jobs on local machine using csh shell
        shell: csh
  system2:
    host: [ login4, login5, login6 ]
    local:
       bash:
        description: submit jobs on local machine using bash shell
        shell: bash
     slurm:
       haswell:
         qos: debug
         cluster: cori
         options:
          - -C haswell
         description: debug queue on Haswell partition
```

The keyword **host** can be used to specify which hostname this executor can run from. If user runs buildtest from `login[1-3]` and uses `system1` they should be able to submit job however if they are on `login4` that job won't be submitted. Similarly, a test using `system2` will need user to login to one of the login nodes and run job from there.

In the buildspec we would need to specify system name to pick the correct executor something like

```
version: "1.0"
buildspecs:
  hostname:
    type: script
    executor: local.bash
    system: system1
    run: hostname
```

**Comment by adityakavalur** [comment]
@shahzebsiddiqui wouldnt this make system a required entry? So lets say a user has n target systems which all run on bash, now they have to create n specs? whereas they didnt before

**Comment by shahzebsiddiqui** [comment]
@adityakavalur i have added support for this but done it slightly different. It's part of the executor property in the format `executor: <site>.<type>.<name>`. You can see https://github.com/buildtesters/buildtest/pull/668 PR which added the change and its part of the docs. All the changes were applied in Cori see https://github.com/buildtesters/buildtest-cori/pull/31 

I will add a new system entry for perlmutter.

**Comment by shahzebsiddiqui** [comment]
Fix #668
